e2e_tests/tests/cluster/test_logging.py

```diff
@@ -105,13 +105,36 @@ def task_log_fields(follow: Optional[bool] = None) -> Iterable[LogFields]:
         return bindings.get_TaskLogsFields(session, taskId=task_id, follow=follow)
 
     try:
-        check_logs(
-            log_regex,
-            functools.partial(api.task_logs, session, task_id),
-            functools.partial(bindings.get_TaskLogsFields, session, taskId=task_id),
-        )
-    except socket.timeout:
-        raise TimeoutError(f"timed out waiting for {task_type} with id {task_id}")
+        result: Any = None
```

Comment:
nit: since the type of result is `Union[bool|Exception]`, could we make it more explicit and call it

```python
exception: Optional[Exception] = None
```

A nit, but one I think makes the flow a tiny bit easier to read.

Reply:
I think I was trying to provide positive confirmation that the thread actually set the value... but I don't check for it anywhere, and I only read it after joining the thread...  So yeah, good idea, I like it that way.